### PR TITLE
[#7] [Chore] [Android] Configure CD for the project

### LINKED ISSUE #7: [Chore] [Android] Configure CD for the project
## Why

To improve production quality, continuous deployment (CD) plays an important part in the development process.

## Acceptance criteria

Set up a CD pipeline that satisfies the development standard.

### PR DESCRIPTION
**Pull request by thiennguyen0196**
- Close #7 

## What happened 👀

- Setup CD workflow for distribution production app to Firebase with [kaylabruce-kmmic](https://console.firebase.google.com/project/kaylabruce-kmmic/overview) app
- Remove BuildKonfig.kt file and let it be generated by the workflow
- Add necessary secret variables `ANDROID_RELEASE_KEYSTORE ` `ANDROID_SIGNING_PROPERTIES ` `FIREBASE_APP_ID_PRODUCTION ` `FIREBASE_DISTRIBUTION_CREDENTIAL_FILE_CONTENT` `GOOGLE_SERVICES_JSON_PRODUCTION` `GOOGLE_SERVICES_JSON_STAGING`

## Insight 📝

The application is distributed (the CD workflow is triggered) whenever the release PR is merged into the `main` branch

## Proof Of Work 📹

This POW is collected by changing the trigger branch of the workflow to the current branch for testing

Screenshot 2023-08-23 at 12 00 32 PM

Screenshot 2023-08-23 at 12 01 42 PM

### CODE REVIEW

File: .github/workflows/android_shared_lint_and_test.yml [hunk 1]
```diff
@@ -32,11 +32,18 @@ jobs:
...
         run: |
-          echo $KMM_KONFIG_PROPERTIES | base64 --decode > buildKonfig.properties
+          echo "$BUILD_KONFIG" > buildSrc/src/main/kotlin/appPackage/BuildKonfig.kt
+
+      - name: Setup Google Services
+        env:
+          GOOGLE_SERVICES_JSON_STAGING: ${{ secrets.GOOGLE_SERVICES_JSON_STAGING }}
+        run: |
+          mkdir -p android/src/staging
+          echo $GOOGLE_SERVICES_JSON_STAGING > android/src/staging/google-services.json
```
Comment (nmint8m) [inline .github/workflows/android_shared_lint_and_test.yml:46]: Please check my suggestion to see if it works (to make it consistent with build konfig)
```suggestion
          echo "$GOOGLE_SERVICES_JSON_STAGING" > android/src/staging/google-services.json
```

File: signing.properties [hunk 2]
```diff
@@ -0,0 +1,3 @@
+KEY_ALIAS=
```
Comment (nmint8m) [inline signing.properties:1]: I would prefer to have an example file instead `signing.properties.example`, then put this file into `.gitignore`

**Review by nmint8m — CHANGES_REQUESTED** [review]

**Comment by thiennguyen0196** [comment]
I have tested and resolved your concerns in this commit https://github.com/nimblehq/ic-kmm-kayla-bruce/pull/86/commits/da2faafabdc7e0158ecf9d80a1685d4c174d3728 @nmint8m 
Please check it again. Many thanks 👏 

Screenshot 2023-08-23 at 7 00 55 PM

**Review by nmint8m — APPROVED** [review]
Look good to me now

**Review by markgravity — APPROVED** [review]

**Pull request merged**